### PR TITLE
Extracted api into mycelium-api crate

### PR DESCRIPTION
**Pull request by maximevanhees**

### CODE REVIEW

**Comment by LeeSmet** [comment]
@maximevanhees it looks like the changes to myceliumd-private aren't included in the commit

**Comment by maximevanhees** [comment]
@LeeSmet, thanks! I've added the files in the latest commit.

File: mycelium-api/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[package]
+name = "mycelium-api"
+version = "0.5.2"
+edition = "2021"
```
Comment (LeeSmet) [inline mycelium-api/Cargo.toml:4]: This is missing the `license-file` and `readme` keys (both can link to the parent directory files, similar to other crates)

**Review by LeeSmet — CHANGES_REQUESTED** [review]
Single comment, additionally the lock file of myceliumd-private is not updated. This can be done by just building the binary and then committing the new lock file.

**Comment by maximevanhees** [comment]
Should be good now.

**Review by LeeSmet — APPROVED** [review]

**Pull request merged**